### PR TITLE
Fixed issues after review.

### PR DESCRIPTION
**Pull request by SonyaPodliesna**

### CODE REVIEW

File: src/app/dashboard/components/challenges/challenges.component.html [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+  <div class="general-content">
+    <div class="main-content">
+      <div class="dot"></div>
+
+      <p class="content-text">Ended</p>
+    </div>
+
+    <div class="content">
+      <div class="event-image" *ngFor="let event of events">
+        <img src="./assets/on-bike.jpg" alt="" class="event-avatar" />
```
Comment (tgogina) [inline src/app/dashboard/components/challenges/challenges.component.html]: do you need empty alt here?

File: src/app/dashboard/components/exoft-achievements/exoft-achievements.component.html [hunk 2]
```diff
@@ -1,28 +1,33 @@
 <div class="exoftachiv">
+
   <div class="tittle">
```
Comment (tgogina) [inline src/app/dashboard/components/exoft-achievements/exoft-achievements.component.html]: `<div class="exoftachiv">` and `<div class="tittle">` are not one-level tags. you can remove epty line here

File: src/app/dashboard/components/exoft-achievements/exoft-achievements.component.html [hunk 3]
```diff
@@ -1,28 +1,33 @@
...
     <span class="name">EXOFT ACHIEVEMENTS</span>
     <div class="achievement-button">{{exoftAchievements.length}}</div>
   </div>
+
   <div class="content">
     <mat-list class="list">
       <mat-list-item *ngFor="let exoftAchievement of exoftAchievements">
         <div class="list-content">
+
           <div class="first-block">
```
Comment (tgogina) [inline src/app/dashboard/components/exoft-achievements/exoft-achievements.component.html]: `<div class="list-content">` and `<div class="first-block"> `are not one-level tags. you can remove epty line here

File: src/app/dashboard/components/greeting/greeting.component.html [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
+
     <div class="header">
-      <span>{{currentTime()}}, <b>{{user.firstName}}</b>!</span>
+      <span>{{currentTime}}, <b>{{user.firstName}}</b>!</span>
     </div>
+
     <div class="sub-header">
       <span class="main-text">Take a look at your activities, standings, friends and more.</span>
     </div>
+
```
Comment (tgogina) [inline src/app/dashboard/components/greeting/greeting.component.html]: 3 and 11 lines - you can remove them :)
please double check all files and formatt html properly. thanks 🙃

File: src/app/dashboard/components/greeting/greeting.component.ts [hunk 5]
```diff
@@ -6,29 +6,28 @@ import { Component, OnInit } from '@angular/core';
   styleUrls: ['./greeting.component.scss']
 })
 export class GreetingComponent implements OnInit {
+  user = { firstName: 'Sonya', lastName: 'Podliesna' };
+  currentTime: string = '';
+
+  constructor() { }
```
Comment (tgogina) [inline src/app/dashboard/components/greeting/greeting.component.ts]: please remove empty constructor

File: src/app/dashboard/components/greeting/greeting.component.ts [hunk 6]
```diff
@@ -6,29 +6,28 @@ import { Component, OnInit } from '@angular/core';
...
 
-  currentTime() {
-    let d = new Date();
-    let n = d.getHours();
+  getCurrentTime() {
```
Comment (tgogina) [inline src/app/dashboard/components/greeting/greeting.component.ts]: provide return type here

File: src/app/dashboard/components/greeting/greeting.component.ts [hunk 7]
```diff
@@ -6,29 +6,28 @@ import { Component, OnInit } from '@angular/core';
...
-
-
-  constructor() { }
-
-  ngOnInit(): void {
+  }
 
+  ngOnInit(): void { 
+    this.currentTime = this.getCurrentTime()
   }
```
Comment (tgogina) [inline src/app/dashboard/components/greeting/greeting.component.ts]: remove empty line

File: src/app/dashboard/components/thank-you/thank-you.component.ts [hunk 8]
```diff
@@ -6,7 +6,8 @@ import { Component, OnInit } from '@angular/core';
   styleUrls: ['./thank-you.component.scss']
 })
 export class ThankYouComponent implements OnInit {
-  public user = { userPic: './assets/user-pic.jpg', firstName: 'Taras', lastName: 'Sava' }
+  user = { userPic: './assets/user-pic.jpg', firstName: 'Taras', lastName: 'Sava' }
+  
   constructor() { }
```
Comment (tgogina) [inline src/app/dashboard/components/thank-you/thank-you.component.ts]: please remove empty consturcotr and ngOnInit

File: src/app/dashboard/components/top-chart/top-chart.component.ts [hunk 9]
```diff
@@ -1,22 +1,19 @@
...
     { userPic: './assets/user-pic.jpg', firstName: 'Tanya', lastName: 'Gogina', xp: 70, color: '#d99842', width: '' },
     { userPic: './assets/user-pic.jpg', firstName: 'Taras', lastName: 'Sava', xp: 15, color: '#78e9e7', width: '' },
     { userPic: './assets/user-pic.jpg', firstName: 'Maksym', lastName: 'Nykolaichuk', xp: 0, color: '#4aa2e0', width: '' }
   ]
 
-
-
   constructor() { }
 
   ngOnInit(): void {
```
Comment (tgogina) [inline src/app/dashboard/components/top-chart/top-chart.component.ts]: same here. remove all unused constructor and ngOnInit functions

File: src/app/dashboard/dashboard.component.scss [hunk 10]
```diff
@@ -1,9 +1,34 @@
...
+
+.badges {
+  height: 100%;
+  width: 100%;
+}
+
+.challenges {
+  height: 100%;
+  width: 100%;
+}
```
Comment (tgogina) [inline src/app/dashboard/dashboard.component.scss]: these classes are duplicated. you can create single common class (.page-block for example) and use it
please change it :)

File: src/app/layout/layout.component.html [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+        </mat-menu>
+      </div>
+    </mat-toolbar>
+
+    <div class="main-content">
+      <router-outlet></router-outlet>
+    </div>
+    
+  </mat-sidenav-content>
+</mat-sidenav-container>
```
Comment (tgogina) [inline src/app/layout/layout.component.html]: Here is an example of formatting. Please take a look and format all html files proparly :)

```suggestion
<mat-sidenav-container fullscreen>
  <mat-sidenav mode="over" #sidenav>
    <app-main-nav (closeSidebar)="sidenav.toggle()"></app-main-nav>
  </mat-sidenav>

  <mat-sidenav-content>
    <mat-toolbar>
      <button mat-icon-button (click)="sidenav.toggle()">
        <mat-icon>menu</mat-icon>
      </button>
      
      GMFCTN!
      
      <span class="toolbar-spacer"></span>
      
      <a mat-button
         class="navigation-item"
         routerLink="app-dashboard"
         routerLinkActive="router-link-active">
         Dashboard
      </a>
         
      <a mat-button  class="navigation-item">Badges</a>
         
      <a mat-button class="navigation-item">Orders</a>
         
      <span class="toolbar-spacer"></span>

      <div class="right-menu">
        <img [src]="user.userPic" alt="user avatar" class="user-pic" />
        
        <button mat-button [matMenuTriggerFor]="menu">
             {{user.firstName}} {{user.lastName}}
             <mat-icon>arrow_drop_down</mat-icon>
        </button>
        
        <mat-menu #menu="matMenu">
          <button mat-menu-item>Edit profile</button>
          
          <button mat-menu-item>Change password</button>
          
          <button mat-menu-item>Log out</button>
        </mat-menu>
      </div>
    </mat-toolbar>

    <div class="main-content">
      <router-outlet></router-outlet>
    </div>
  </mat-sidenav-content>
</mat-sidenav-container>
```

File: src/app/main-nav/main-nav.component.ts [hunk 12]
```diff
@@ -1,15 +1,43 @@
...
+    { pic: './assets/achievement-icon.png', name: 'Exoft Turbo Power', xp: 15 },
+    { pic: './assets/achievement-icon.png', name: 'Exoft Turbo Power', xp: 15 },
+    { pic: './assets/achievement-icon.png', name: 'Exoft Skylark Power', xp: 20 },
+    { pic: './assets/achievement-icon.png', name: 'Exoft Corpo Power', xp: 20 }
+  ];
+
+  totalXP: number = 0;
+
+  constructor() {
+  }
```
Comment (tgogina) [inline src/app/main-nav/main-nav.component.ts]: empty constructor 🙃
remove it

File: src/app/main-nav/main-nav.component.ts [hunk 13]
```diff
@@ -1,15 +1,43 @@
...
+    });
+
+    return counter;
   }
 
+  hideSidebar() {
+    this.closeSidebar.emit();
+  }
+
+
```
Comment (tgogina) [inline src/app/main-nav/main-nav.component.ts]: remove these empty lines

File: src/app/shared/pipes/difference-in-time.pipe.ts [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+//   transform(value: Date): string {
+//     const difference = Math.abs(new Date().getTime() - new Date(value).getTime());
+//     const diffInMinutes = difference / 1000 / 60 / 60;
+    
+//     if (diffInMinutes >= 60) {
+//       return `${Math.round(diffInMinutes / 60)} hours ago`;
+//     } 
+//       return `${diffInMinutes} mins ago`;
+//   }
```
Comment (tgogina) [inline src/app/shared/pipes/difference-in-time.pipe.ts]: please remove commented code

File: src/app/shared/pipes/difference-in-time.pipe.ts [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+
+transform(value: Date): string {
+  const difference = Math.abs(new Date().getTime() - new Date(value).getTime());
+  const diffInMinutes = difference / 1000 / 60;
+  console.log(diffInMinutes);
```
Comment (tgogina) [inline src/app/shared/pipes/difference-in-time.pipe.ts]: console.log and debuggers are unacceptable to be present in PR. please always check your code and do not keep console.log and debuggers in your app.

File: src/app/dashboard/components/challenges/challenges.component.html (hunk 1 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/components/challenges/challenges.component.html]: fixed

File: src/app/dashboard/components/exoft-achievements/exoft-achievements.component.html (hunk 2 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/components/exoft-achievements/exoft-achievements.component.html]: fixed

File: src/app/dashboard/components/exoft-achievements/exoft-achievements.component.html (hunk 3 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/components/exoft-achievements/exoft-achievements.component.html]: fixed

File: src/app/dashboard/components/greeting/greeting.component.html (hunk 4 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/components/greeting/greeting.component.html]: fixed

File: src/app/dashboard/components/greeting/greeting.component.ts (hunk 5 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/components/greeting/greeting.component.ts]: fixed

File: src/app/dashboard/components/greeting/greeting.component.ts (hunk 6 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/components/greeting/greeting.component.ts]: fixed

File: src/app/dashboard/components/greeting/greeting.component.ts (hunk 7 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/components/greeting/greeting.component.ts]: fixed

File: src/app/dashboard/components/thank-you/thank-you.component.ts (hunk 8 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/components/thank-you/thank-you.component.ts]: fixed

File: src/app/dashboard/components/top-chart/top-chart.component.ts (hunk 9 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/components/top-chart/top-chart.component.ts]: fixed

File: src/app/dashboard/dashboard.component.scss (hunk 10 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/dashboard.component.scss]: fixed

File: src/app/layout/layout.component.html (hunk 11 above)
Comment (SonyaPodliesna) [reply src/app/layout/layout.component.html]: fixed

File: src/app/main-nav/main-nav.component.ts (hunk 12 above)
Comment (SonyaPodliesna) [reply src/app/main-nav/main-nav.component.ts]: fixed

File: src/app/main-nav/main-nav.component.ts (hunk 13 above)
Comment (SonyaPodliesna) [reply src/app/main-nav/main-nav.component.ts]: fixed

File: src/app/shared/pipes/difference-in-time.pipe.ts (hunk 14 above)
Comment (SonyaPodliesna) [reply src/app/shared/pipes/difference-in-time.pipe.ts]: fixed

File: src/app/shared/pipes/difference-in-time.pipe.ts (hunk 15 above)
Comment (SonyaPodliesna) [reply src/app/shared/pipes/difference-in-time.pipe.ts]: fixed

**Review by tgogina — APPROVED** [review]

**Pull request merged**